src/extractor/edge_based_graph_factory.cpp

```diff
@@ -286,15 +299,20 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedNodes()
 
                 BOOST_ASSERT(node_u < node_v);
 
+                boost::optional<Mapping> mapping;
+
                 // if we found a non-forward edge reverse and try again
                 if (edge_data.edge_id == SPECIAL_NODEID)
                 {
-                    InsertEdgeBasedNode(node_v, node_u);
+                    mapping = InsertEdgeBasedNode(node_v, node_u);
```

Comment:
hm. This `mapping` is kind of hanging in the air for me. @daniel-j-h do you see a way we could do this in a less `hacky` approach? Interleaving it with the EdgeBasedGraph Generation seems a bit messy.